client/web/compose/src/views/Admin/Modules/Edit.vue

```diff
@@ -840,6 +846,49 @@ export default {
         })
     },
 
+    handleClone () {
+      const module = this.module.clone()
+      module.moduleID = NoID
+      module.name = `${this.module.name} (copy)`
+      module.handle = `${this.module.handle}_copy`
+
+      const resourceTranslationLanguage = this.currentLanguage
```

Comment:
Same here, instead of copying the code just call handleSave(module) and make it accept a module.
Also make sure it doesn't check for isEdit, but instead it check if ID is equal NoID, like chart does.